### PR TITLE
Use ESM in demo

### PR DESCRIPTION
**Pull request by xtuc**
**What**:

**Why**:

Fixes https://github.com/tkh44/emotion/pull/160

**How**:

Enabling ESM in Emotion and in build process.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Code complete

@lukeed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "6.0.0",
   "description": "high performance css-in-js",
   "main": "lib/index.js",
+  "module": "src/",
```
Comment (xtuc) [inline package.json]: I took a shortcut here. I don't know how you want to handle the ESM build of Emotion.

File: .babelrc [hunk 2]
```diff
@@ -3,10 +3,10 @@
     [
       "env",
       {
-        "loose": true
+        "loose": true,
+        "modules": false,
       }
     ],
-    "stage-0",
```
Comment (xtuc) [inline .babelrc]: stage-0 wasn't necessary, I would recommend you to avoiding relying on stage-* presets. Use a [plugins](https://babeljs.io/docs/plugins/#transform-plugins-es2015) directly.

**Comment by lukeed** [comment]
Thanks @xtuc!

Commenting as I run this locally:

* This [line from `macro-styled`](https://github.com/xtuc/emotion/blob/de645271d88e3ac4301610ff61177655911139ff/src/macro-styled.js#L54) has to be fixed

**Comment by xtuc** [comment]
You can ping me in future PRs about that, I'll take a look.

**Comment by tkh44** [comment]
Thanks @xtuc!

**Pull request closed without merging**